superset/models/core.py

```diff
@@ -510,12 +511,22 @@ def _get_sqla_engine(  # pylint: disable=too-many-locals
         logger.debug("Database._get_sqla_engine(). Masked URL: %s", str(masked_url))
 
         if self.impersonate_user:
-            self.db_engine_spec.update_impersonation_config(
-                connect_args,
-                str(sqlalchemy_url),
-                effective_username,
-                access_token,
-            )
+            # Checking if the function signature can accept database as a param
```

Comment:
Again, here BigQuery needs the Database object to build the client. Checks if the function signature has `database` as a param, then passes it.